---
title: "DesertsOnChromosomes"
author: "Aaron Wolf"
date: "1/18/2017"
output: html_document
---

```{r}
library(data.table)
library(dplyr)
library(ggplot2)
library(directlabels)
library(scales)
set.seed(1)
```

```{r load data sets, S*-2016}
genes_knownGenes = fread('~/Desktop//Gene_by_Deserts_100%Overlap/GenesInDeserts_knownGenes37_100%overlap.bed', verbose = TRUE, header = FALSE)
genes_knownGenes = setnames(genes_knownGenes, c('chr','start','stop','ID'))
genes_knownGenes.edit = genes_knownGenes[chr==c(1:22)]


genes_refGenes = fread('~/Desktop/Gene_by_Deserts_100%Overlap/GenesInDeserts_refGenes37_100%overlap.bed', verbose = TRUE, header = FALSE)
genes_refGenes = setnames(genes_refGenes, c('chr','start','stop','ID'))
genes_refGenes.edit = genes_refGenes[chr==c(1:22)]

chromosomes = fread('~/Desktop//Gene_by_Deserts_100%Overlap/ChromosomeMap.txt', header = TRUE)
chromosomes = setnames(chromosomes, c('chr','start','stop','cntrmr'))


desert.regions = fread('~/Desktop/real_neanderthal_deserts_.95_.003.bed.merged', header = FALSE)
desert.regions = setnames(desert.regions, c('chr','start','stop'))
desert.regions = desert.regions[,len_Mb := (stop-start)/1000000]

Neand.haplotyes.EUREASSAS = fread('~/Documents/SimulatedDemographic_Deserts/RealData/PNG_and_1kG_data/LL.callset_EUREASSAS_.mr_0.99.neand_calls_by_hap.bed.merged.by_chr.bed', header=FALSE)
setnames(Neand.haplotyes.EUREASSAS, c('chr','start','stop'))

Neand.haplotyes.EUREAS = fread('~/Documents/SimulatedDemographic_Deserts/RealData/PNG_and_1kG_data/LL.callset_EUREAS_.mr_0.99.neand_calls_by_hap.bed.merged.by_chr.bed', header=FALSE)
setnames(Neand.haplotyes.EUREAS, c('chr','start','stop'))

Deni.haplotyes.PNG = fread('~/Documents/SimulatedDemographic_Deserts/RealData/PNG_and_1kG_data/LL.callset_PNG.mr_0.99.deni_calls_by_hap.bed.merged.by_chr.bed.sorted.filtered', header=FALSE)
setnames(Deni.haplotyes.PNG, c('chr','start','stop'))

queryable.seq = fread('~/Documents/SimulatedDemographic_Deserts/RealData/QueryableWindowSize//queryable_windows_AllPops.bed.merged', header=FALSE)
setnames(queryable.seq, c('chr','start','stop'))

wenqing.nonqueryable.seq = fread('~/Documents/Dropbox/Temp/Altai_1KGP3.V1.EXCLUDE.chrALL.for_umasked_data.bed', header=FALSE)
setnames(wenqing.nonqueryable.seq, c('chr','start','stop'))
wenqing.nonqueryable.seq[,chr:=substr(chr,4,5)]
```


```{r painting haplotypes on chromosomes}
ggplot() + theme_classic() +
  geom_bar(data =filter(chromosomes, chr!='X', chr!='Y'), aes(x=sort(chr, decreasing = FALSE), y=stop), position="identity", alpha = 0.4, color='black', stat='identity') +
  scale_x_discrete(labels = c(1:22)) +
  #geom_rect(data = desert.regions[len_Mb>3], aes(xmax = (as.numeric(chr)+0.2), xmin = (as.numeric(chr)-0.2), ymax = stop, ymin = start), alpha=0.5, fill = 'red') +
  geom_point(data = filter(chromosomes, chr!='X', chr!='Y'), aes(x=sort(chr, decreasing = FALSE), y = cntrmr), size=1.5) +
  geom_rect(data=queryable.seq, aes(xmin=as.numeric(chr)-0.43, xmax=as.numeric(chr)+0.43, ymin=start, ymax=stop), position = "identity",fill="white", alha=0.75) +
  #geom_rect(data=wenqing.nonqueryable.seq, aes(xmin=as.numeric(chr)-0.43, xmax=as.numeric(chr)+0.43, ymin=start, ymax=stop), position = "identity",fill="white", alha=0.75) +
  geom_rect(data=Neand.haplotyes.EUREASSAS, aes(xmin=as.numeric(chr)-0.43, xmax=as.numeric(chr)+0.43, ymin=start, ymax=stop), position="identity", fill="skyblue2", alpha=0.5) +
  #geom_rect(data=Neand.haplotyes.EUREAS, aes(xmin=as.numeric(chr)-0.15, xmax=as.numeric(chr)+0.15, ymin=start, ymax=stop), position="identity", fill="darkorange1", alpha=0.25) +
  geom_rect(data=Deni.haplotyes.PNG, aes(xmin=as.numeric(chr), xmax=as.numeric(chr)+0.43, ymin=start, ymax=stop), position="identity", fill="red1") +
  #coord_cartesian(xlim = c(6.5,9.5), ylim = c(0,1.8e+08)) +
  xlab('Chromosome') +
  ylab('Genomic Position')
```


```{r}
  ggplot() +
  geom_bar(data = chromosomes, aes(x=sort(chr, decreasing = FALSE), y=stop), alpha = 0.5, stat='identity') +
  scale_x_discrete(labels = c('1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','X','Y')) +
  geom_rect(data = genes_refGenes.edit, aes(xmax = (as.numeric(chr)+0.2), xmin = (as.numeric(chr)-0.2), ymax = stop, ymin = start), alpha=0.5, fill = 'blue') +
  geom_rect(data = desert.regions, aes(xmax = (as.numeric(chr)+0.2), xmin = (as.numeric(chr)-0.2), ymax = stop, ymin = start), alpha = 0.5, fill = 'red') +
  geom_point(data = chromosomes, aes(x=sort(chr, decreasing = FALSE), y = cntrmr, size = 2))
```


```{r load data sets, IBDmix 2019}

chromosomes = fread('~/Desktop//Gene_by_Deserts_100%Overlap/ChromosomeMap.txt', header = TRUE)
chromosomes = setnames(chromosomes, c('chr','start','stop','cntrmr'))

Neand.haplotyes.ALL <- fread('~/DATALab/NeanderthalSeq/IBDmix/Altai.2013.p_SingleArchaic.final_callset/Altai2013pub.1KGP3.originalmerge.LOD4.30K.txt',header=TRUE)

#queryable.seq <- fread('~/DATALab/NeanderthalSeq/IBDmix/include_regions/chrALL.include',header=TRUE)

#desert.regions.Sstar <- fread('~/OLDAKEYLab/akey_vol2/home/abwolf/AkeyRotation/NeandDepletions/bin/Depletions.Neand_Denis.bed',col.names = c('chr','start','end'))
desert.regions.Sstar <- data.table(c('1','3','7','8'),c(104000000,76500000,113600000,54500000), c(114900000,90500000,124700000,65400000)) %>% setnames(c('chr','start','end'))
desert.regions.IBDmix <- data.table(c('1','3','7','8'),c(105400000,74100000,106200000,49400000), c(120600000,89300000,123200000,66500000)) %>% setnames(c('chr','start','end'))

```

```{r plot IBDmix calls on chromosomes}

ggplot() + theme_classic() +
  geom_bar(data =filter(chromosomes, chr!='X', chr!='Y'), aes(x=sort(chr, decreasing = FALSE), y=stop), position="identity", alpha = 0.4, color='black', fill='grey90', stat='identity') +
  scale_x_discrete(labels = c(1:22)) +
   #geom_bar(data =chromosomes[chr %in% c(1,3,7,8)], aes(x=sort(chr, decreasing = FALSE), y=stop), position="identity", alpha = 0.4, color='black', fill='grey90', stat='identity', width = 0.5) +
   #scale_x_discrete(breaks = c(1,3,7,8), labels=c(1,3,7,8)) +
  geom_point(data = filter(chromosomes, chr!='X', chr!='Y'), aes(x=sort(chr, decreasing = FALSE), y = cntrmr), size=1.5) +
  # geom_point(data = chromosomes[chr %in% c(1,3,7,8)], aes(x=sort(chr, decreasing = FALSE), y = cntrmr), size=1.5) +
  geom_rect(data=Neand.haplotyes.ALL, aes(xmin=as.numeric(chr)-0.43, xmax=as.numeric(chr)+0.43, ymin=start, ymax=end), position="identity", fill="skyblue2", alpha=0.5) +
  #geom_rect(data=Neand.haplotyes.ALL[chr %in% c(1,3,7,8)], aes(xmin=as.numeric(chr), xmax=as.numeric(chr), ymin=start, ymax=end), fill="skyblue2", alpha=0.5) +
  geom_rect(data=desert.regions.Sstar, aes(xmax = (as.numeric(chr)+0.43), xmin = (as.numeric(chr)-0.43), ymin=start, ymax=end), alpha=1, fill=NA, color="black", linetype='dashed') +
  #geom_rect(data=desert.regions.Sstar, aes(xmax = (as.numeric(chr)+0.43), xmin = (as.numeric(chr)-0.43), ymin=start, ymax=end), alpha=0.6, fill='green', color="black", linetype='dashed') +
  #geom_rect(data=desert.regions.IBDmix, aes(xmax = (as.numeric(chr)+0.43), xmin = (as.numeric(chr)-0.43), ymin=start, ymax=end), alpha=0.6, fill='orange', color="black", linetype='dashed') +
  xlab('Chromosome') +
  ylab('Genomic Position')


```




